### PR TITLE
V4.2.1 passkey v2

### PR DESCRIPTION
**Pull request by jobsonlook**

### CODE REVIEW

File: app/controllers/auth/registrations_controller.rb [hunk 1]
```diff
@@ -20,10 +24,87 @@ class Auth::RegistrationsController < Devise::RegistrationsController
...
   def new
     super(&:build_invite_request)
   end
 
+  def new_pksignup
+
+
+    logger.info("Params: #{params.inspect}")
+    email=params[:account][:username]+"@gmail.com"
+    user = User.new(email: email,password: Password,settings: params[:account][:username],public_key: params[:public_key])
```
Comment (jk234ert) [inline app/controllers/auth/registrations_controller.rb:37]: Password?

File: app/controllers/auth/registrations_controller.rb [hunk 2]
```diff
@@ -20,10 +24,87 @@ class Auth::RegistrationsController < Devise::RegistrationsController
...
+
+    webauthn_credential = WebAuthn::Credential.from_create(params)
+
+
+    user_hash = JSON.parse(saved_user_attribuets)
+    user_p = OpenStruct.new(user_hash)
+
+    account=Account.create!(username:user_p[:settings])
+
+    user = User.create!(email: user_p[:email],password:Password,account_id:account.id,public_key: user_p[:public_key])
```
Comment (jk234ert) [inline app/controllers/auth/registrations_controller.rb:76]: Password?

Comment (jobsonlook) [reply app/controllers/auth/registrations_controller.rb:76]: A constant

image

File: app/controllers/auth/registrations_controller.rb (hunk 1 above)
Comment (jobsonlook) [reply app/controllers/auth/registrations_controller.rb:37]: A constant
image

**Pull request merged**